presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -293,7 +298,29 @@ public IPrestoSparkQueryExecution create(
         }
         else {
             checkArgument(sqlLocation.isPresent(), "sqlText or sqlLocation must be present");
-            sql = new String(metadataStorage.read(sqlLocation.get()), UTF_8);
+            byte[] sqlFileBytes = metadataStorage.read(sqlLocation.get());
+            if (sqlFileSizeInBytes.isPresent()) {
+                checkArgument(
```

Comment:
Let's classify these errors (as user error?)

Reply:
Actually user didn't set this argument,  the query file could be changed due to transport error, malicious attack etc. How about classify it as PrestoException but with error code in category SparkErrorCode?  Alerting may ignore user error category.

Reply:
Sounds good